src/components/Node/Node.wrapper.tsx

```diff
@@ -1,25 +1,95 @@
 import * as React from 'react'
 import Draggable from 'react-draggable'
-import { INode, IOnDragNode, IOnNodeClick, ISelectedOrHovered } from '../../'
+import {
+  ILink, INode, INodeInnerDefaultProps, IOnDragNode,
+  IOnLinkCancel, IOnLinkComplete, IOnLinkMove, IOnLinkStart,
+  IOnNodeClick, IOnNodeSizeChange, IOnPortPositionChange,
+  IPortDefaultProps, IPortsDefaultProps, IPosition, ISelectedOrHovered, ISize, PortWrapper,
+} from '../../'
 import { INodeDefaultProps, NodeDefault } from './Node.default'
 
 export interface INodeWrapperProps {
   node: INode
+  Component: React.ElementType<INodeDefaultProps>
+  offset: IPosition
+  selected: ISelectedOrHovered | undefined
+  hovered: ISelectedOrHovered | undefined
+  selectedLink: ILink | undefined
+  hoveredLink: ILink | undefined
+  isSelected: boolean
+  NodeInner: React.ElementType<INodeInnerDefaultProps>
+  Ports: React.ElementType<IPortsDefaultProps>
+  Port: React.ElementType<IPortDefaultProps>
+  onPortPositionChange: IOnPortPositionChange
+  onLinkStart: IOnLinkStart
+  onLinkMove: IOnLinkMove
+  onLinkComplete: IOnLinkComplete
+  onLinkCancel: IOnLinkCancel
   onDragNode: IOnDragNode
   onNodeClick: IOnNodeClick
-  children: any
-  selected: ISelectedOrHovered
-  Component?: React.SFC<INodeDefaultProps>
+  onNodeSizeChange: IOnNodeSizeChange
 }
 
 export const NodeWrapper = ({
   node,
   onDragNode,
-  children,
   onNodeClick,
-  selected,
+  isSelected,
   Component = NodeDefault,
+  onNodeSizeChange,
+  NodeInner,
+  Ports,
+  Port,
+  offset,
+  selected,
+  selectedLink,
+  hovered,
+  hoveredLink,
+  onPortPositionChange,
+  onLinkStart,
+  onLinkMove,
+  onLinkComplete,
+  onLinkCancel,
 }: INodeWrapperProps) => {
+  const [size, setSize] = React.useState<ISize>({ width: 0, height: 0 })
+  const compRef = React.useRef<HTMLElement>(null)
+  React.useEffect(() => {
```

Comment:
effect on every state value change? may be define it right away?
```
useEffect(() => {...}, [size])
```
?

Reply:
Effect is here to prevent state changing while rendering, but yeah, it doesn't look right actually. Still new at all this hooks stuff. I'll figure it out.